pkg/executor/check_table_index.go

```diff
@@ -508,37 +549,21 @@ func (w *checkIndexWorker) HandleTask(task checkIndexTask, _ func(workerpool.Non
 		mod *= bucketSize
 	}
 
-	queryToRow := func(se sessionctx.Context, sql string) ([]chunk.Row, error) {
-		rs, err := se.GetSQLExecutor().ExecuteInternal(ctx, sql)
-		if err != nil {
-			return nil, err
-		}
-		row, err := sqlexec.DrainRecordSet(ctx, rs, 4096)
-		if err != nil {
-			return nil, err
-		}
-		err = rs.Close()
-		if err != nil {
-			logutil.BgLogger().Warn("close result set failed", zap.Error(err))
-		}
-		return row, nil
-	}
-
 	if meetError {
 		groupByKey := fmt.Sprintf("((cast(%s as signed) - %d) %% %d)", md5Handle, offset, mod)
 		indexSQL := fmt.Sprintf(
-			"select %s, %s, %s from %s use index(`%s`) where %s = 0 order by %s",
+			"select /*+ AGG_TO_COP() */ %s, %s, %s from %s use index(`%s`) where %s = 0 order by %s",
 			handleColumns, indexColumns, md5HandleAndIndexCol, tblName, idxInfo.Name, groupByKey, handleColumns)
 		tableSQL := fmt.Sprintf(
-			"select /*+ read_from_storage(tikv[%s]) */ %s, %s, %s from %s use index() where %s = 0 order by %s",
+			"select /*+ read_from_storage(tikv[%s]), AGG_TO_COP() */ %s, %s, %s from %s use index() where %s = 0 order by %s",
 			tblName, handleColumns, indexColumns, md5HandleAndIndexCol, tblName, groupByKey, handleColumns)
-
-		idxRow, err := queryToRow(se, indexSQL)
+		intest.Assert(verifyIndexSideQuery(ctx, se, indexSQL), "index side query plan is not correct: %s", indexSQL)
```

Comment:
I think `AssertFunc` is better than `Assert`. `Assert` will always evaluate `verifyIndexSideQuery(ctx, se, indexSQL)` then check the boolean result even if `intest` tag is not set.